core/src/main/java/org/elasticsearch/index/query/PrefixQueryBuilder.java

```diff
@@ -19,20 +19,31 @@
 
 package org.elasticsearch.index.query;
 
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.MultiTermQuery;
+import org.apache.lucene.search.PrefixQuery;
+import org.apache.lucene.search.Query;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.lucene.BytesRefs;
 import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.index.mapper.MappedFieldType;
+import org.elasticsearch.index.query.support.QueryParsers;
 
 import java.io.IOException;
+import java.util.Objects;
 
 /**
  * A Query that matches documents containing terms with a specified prefix.
  */
-public class PrefixQueryBuilder extends MultiTermQueryBuilder {
+public class PrefixQueryBuilder extends MultiTermQueryBuilder<PrefixQueryBuilder> {
 
     public static final String NAME = "prefix";
 
-    private final String name;
+    private final String fieldName;
 
-    private final String prefix;
+    private final Object value;
```

Comment:
We definitely could. On a different note I was thinking about making PrefixQueryBuilder extends from BaseTermQueryBuilder, now that MultiTermQueryBuilder is an interface. Sounds good?